src/order/closure.lean

```diff
@@ -112,12 +127,45 @@ set.ext $ λ x, ⟨λ h, ⟨x, h⟩, by { rintro ⟨y, rfl⟩, apply c.idempoten
 /-- Send an `x` to an element of the set of closed elements (by taking the closure). -/
 def to_closed (x : α) : c.closed := ⟨c x, c.closure_is_closed x⟩
 
+lemma mk₃.to_closed {f : α → α} {p : α → Prop} {hf : ∀ x, x ≤ f x} {hfp : ∀ x, p (f x)}
+  {hmin : ∀ ⦃x y⦄, x ≤ y → p y → f x ≤ y} {x : α} {hx : p x} :
+  (mk₃ f p hf hfp hmin).closed x :=
+le_antisymm (hmin (le_refl _) hx) (hf _)
+
 lemma top_mem_closed {α : Type u} [order_top α] (c : closure_operator α) : ⊤ ∈ c.closed :=
 c.closure_top
 
 lemma closure_le_closed_iff_le {x y : α} (hy : c.closed y) : x ≤ y ↔ c x ≤ y :=
 by rw [← c.closure_eq_self_of_mem_closed hy, le_closure_iff]
 
+lemma closure_closure_sup_left {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (c x ⊔ y) = c (x ⊔ y) :=
+le_antisymm ((closure_le_closed_iff_le c (closure_is_closed _ _)).1 (sup_le (c.monotone le_sup_left)
+  (le_trans le_sup_right (le_closure _ _)))) (c.monotone (sup_le_sup_right (le_closure _ _) _))
+
+lemma closure_closure_sup_right {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (x ⊔ c y) = c (x ⊔ y) :=
+by { rw [sup_comm, closure_closure_sup_left, sup_comm] }
+
+lemma closure_closure_sup {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (c x ⊔ c y) = c (x ⊔ y) :=
+by { rw [closure_closure_sup_left, closure_closure_sup_right] }
+
+lemma closure_closure_supr {α : Type u} {ι : Type v} [complete_lattice α] (c : closure_operator α)
```

Comment:
I thought I already commented this but I can't see the comment:
```suggestion
lemma closure_closure_supr {α : Type u} {ι : Sort v} [complete_lattice α] (c : closure_operator α)
```

Reply:
Oops sorry, I forgot to change that